src/systems/super-spawner-system.js

```diff
@@ -1,53 +1,75 @@
-        options.entity.object3D.position,
-        options.entity.object3D.quaternion,
-        options.entity.object3D.scale
-      );
-      const data = superSpawner.data;
-      const entity = addMedia(data.src, data.template, ObjectContentOrigins.SPAWNER, data.resolve, data.resize, false)
-        .entity;
-      entity.object3D.position.copy(
-        data.useCustomSpawnPosition ? data.spawnPosition : superSpawner.el.object3D.position
-      );
-      entity.object3D.rotation.copy(
-        data.useCustomSpawnRotation ? data.spawnRotation : superSpawner.el.object3D.rotation
-      );
-      entity.object3D.scale.copy(data.useCustomSpawnScale ? data.spawnScale : superSpawner.el.object3D.scale);
-      entity.object3D.matrixNeedsUpdate = true;
-      state.held = entity;
-
-      superSpawner.activateCooldown();
-      state.spawning = true;
-      // WARNING: waitForEvent is semantically different than entity.addEventListener("body-loaded", ...)
-      // and adding a callback fn via addEventListener will not work unless the callback function
-      // wraps its code in setTimeout(()=>{...}, 0)
-      await waitForEvent("body-loaded", entity);
-      state.spawning = false;
-      entity.object3D.position.copy(
-        data.useCustomSpawnPosition ? data.spawnPosition : superSpawner.el.object3D.position
-      );
-      if (data.centerSpawnedObject) {
-        entity.body.position.copy(options.entity.object3D.position);
-      }
-      entity.object3D.matrixNeedsUpdate = true;
+    if (superSpawner && !superSpawner.cooldownTimeout && userinput.get(grabPath)) {
+      this.performSpawn(state, entity, grabPath, userinput, superSpawner);
     }
   }
 
+  performSpawn(state, entity, grabPath, userinput, superSpawner) {
+    entity.object3D.updateMatrices();
+    entity.object3D.matrix.decompose(entity.object3D.position, entity.object3D.quaternion, entity.object3D.scale);
+    const data = superSpawner.data;
+    const spawnedEntity = addMedia(
+      data.src,
+      data.template,
+      ObjectContentOrigins.SPAWNER,
+      data.resolve,
+      data.resize,
+      false
+    ).entity;
+    spawnedEntity.object3D.position.copy(
+      data.useCustomSpawnPosition ? data.spawnPosition : superSpawner.el.object3D.position
+    );
+    spawnedEntity.object3D.rotation.copy(
+      data.useCustomSpawnRotation ? data.spawnRotation : superSpawner.el.object3D.rotation
+    );
+    spawnedEntity.object3D.scale.copy(data.useCustomSpawnScale ? data.spawnScale : superSpawner.el.object3D.scale);
+    spawnedEntity.object3D.matrixNeedsUpdate = true;
+    state.held = spawnedEntity;
+
+    superSpawner.activateCooldown();
+    state.spawning = true;
+    // WARNING: waitForEvent is semantically different than entity.addEventListener("body-loaded", ...)
+    // and adding a callback fn via addEventListener will not work unless the callback function
+    // wraps its code in setTimeout(()=>{...}, 0)
+    spawnedEntity.addEventListener(
```

Comment:
If this works correctly that's new (and good). Delete the comment above warning against doing this.

Reply:
Ah totally missed that comment. What's a test case? (This PR is still WIP, haven't done a lot of testing yet)